baybe/surrogates/gaussian_process/kernel_factory.py

```diff
@@ -0,0 +1,55 @@
+"""Kernel factories for the Gaussian process surrogate."""
+
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Protocol, Union
+
+from attrs import define, field
+from attrs.validators import instance_of
+
+from baybe.kernels.base import Kernel
+from baybe.searchspace import SearchSpace
+from baybe.serialization.core import (
+    converter,
+    get_base_structure_hook,
+    unstructure_base,
+)
+from baybe.serialization.mixin import SerialMixin
+
+if TYPE_CHECKING:
+    from torch import Tensor
+
+
+class KernelFactory(Protocol):
+    """A protocol defining the interface expected for kernel factories."""
+
+    def __call__(
+        self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
+    ) -> Kernel:
+        """Create a :class:`baybe.kernels.base.Kernel` for the given context."""
+        ...
+
+
+# Register de-/serialization hooks
+converter.register_structure_hook(KernelFactory, get_base_structure_hook(KernelFactory))
+converter.register_unstructure_hook(KernelFactory, unstructure_base)
+
+
+@define(frozen=True)
+class PlainKernelFactory(KernelFactory, SerialMixin):
+    """A trivial factory that returns a fixed pre-defined kernel upon request."""
+
+    kernel: Kernel = field(validator=instance_of(Kernel))
+    """The fixed kernel to be returned by the factory."""
+
+    def __call__(  # noqa: D102
+        self, searchspace: SearchSpace, train_x: Tensor, train_y: Tensor
+    ) -> Kernel:
+        # See base class.
+
+        return self.kernel
+
+
+def to_kernel_factory(x: Union[Kernel, KernelFactory], /) -> KernelFactory:
```

Comment:
alternative suggestion: instead of making this a utility located here is there a way to give the abstract `Kernel` class a `to_factory` method? Would seem a bit more convenient to me

Reply:
Yes, could to so (and theoretically keep both), but the reason why I added it as a converter is to make users' lifes easier. In fact, you [noticed](https://github.com/emdgroup/baybe/pull/235#discussion_r1601115072) the complication that would arise without it yourself: From a technical reason (which I'll explain in the other thread), we need a factory here, but I think people would often want to simply just pass a regular Kernel instead. The converter takes care of that, which is especially useful for JSON settings. (Note, this is analogous to the target-to-objective converter in campaign, and also to a subspacediscrete/subspacecontinuous-to-searchspace converter we could add for searchspaces). I was also thinking about whether we should expose this field as "kernel_or_factory" to make the conversion explicit but haven't done so far, also because it's not the case campaign...

Long story short:
* if you want, I can add the `to_factory` method, also just for consistency with the `to_objective` method of targets (let me know)
* however, if I removed the converter, people would have a harder time writing the JSON

Reply:
ok i see the point for having it as a converter and its importance for json, i didnt necessarily mean to suggest to remove it
having said that a `to_factory` would be nice